### PR TITLE
Bump golangci-lint to v1.50.1

### PR DESCRIPTION
**Pull request by lucacome**
Updates:
- ~~golangci-lint aligning CI and Makefile~~
- golangci/golangci-lint-action to `v3.3.1`
- actions/setup-go to `v3` to be able to read Go version from `go.mod`

### CODE REVIEW

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -18,13 +18,13 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v3
       - name: install Go
-        uses: actions/setup-go@v2
+        uses: actions/setup-go@v3
         with:
-          go-version: 1.18.x
+          go-version-file: go.mod
```
Comment (mem) [inline .github/workflows/golangci-lint.yml:23]: Looking at what the action does, I don't think this is going to have any effect _except_ if ruleguard is configured. It should be OK.

**Review by mem — COMMENTED** [review]
looks OK to me.

**Comment by roidelapluie** [comment]
makefile.common comes from prometheus/prometheus and should be update here

**Comment by lucacome** [comment]
@roidelapluie done https://github.com/prometheus/prometheus/pull/11758

**Comment by lucacome** [comment]
@roidelapluie can it be merged?

**Comment by SuperQ** [comment]
Fixed in #447

**Pull request closed without merging**